packages/block-editor/src/hooks/color.js

```diff
@@ -0,0 +1,250 @@
+	getColorClassName,
+	getColorObjectByColorValue,
+	getColorObjectByAttributeValues,
+} from '../components/colors';
+import PanelColorSettings from '../components/panel-color-settings';
+import ContrastChecker from '../components/contrast-checker';
+import InspectorControls from '../components/inspector-controls';
+import { getBlockDOMNode } from '../utils/dom';
+
+export const COLOR_SUPPORT_KEY = '__experimentalColor';
+
+/**
+ * Filters registered block settings, extending attributes to include
+ * `backgroundColor` and `textColor` attribute.
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+function addAttributes( settings ) {
+	if ( ! hasBlockSupport( settings, COLOR_SUPPORT_KEY ) ) {
+		return settings;
+	}
+
+	// allow blocks to specify their own attribute definition with default values if needed.
+	if ( ! settings.attributes.backgroundColor ) {
+		Object.assign( settings.attributes, {
+			backgroundColor: {
+				type: 'string',
+			},
+		} );
+	}
+	if ( ! settings.attributes.textColor ) {
+		Object.assign( settings.attributes, {
+			textColor: {
+				type: 'string',
+			},
+		} );
+	}
+
+	return settings;
+}
+
+/**
+ * Override props assigned to save component to inject colors classnames.
+ *
+ * @param  {Object} props      Additional props applied to save element
+ * @param  {Object} blockType  Block type
+ * @param  {Object} attributes Block attributes
+ * @return {Object}            Filtered props applied to save element
+ */
+export function addSaveProps( props, blockType, attributes ) {
+	if ( ! hasBlockSupport( blockType, COLOR_SUPPORT_KEY ) ) {
+		return props;
+	}
+
+	// I'd have prefered to avoid the "style" attribute usage here
+	const { backgroundColor, textColor, style } = attributes;
+
+	const backgroundClass = getColorClassName(
+		'background-color',
+		backgroundColor
+	);
+	const textClass = getColorClassName( 'color', textColor );
+	props.className = classnames( props.className, backgroundClass, textClass, {
+		'has-text-color': textColor || style?.color?.text,
+		'has-background': backgroundColor || style?.color?.background,
+	} );
+
+	return props;
+}
+
+/**
+ * Filters registered block settings to extand the block edit wrapper
+ * to apply the desired styles and classnames properly.
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+export function addEditProps( settings ) {
+	if ( ! hasBlockSupport( settings, COLOR_SUPPORT_KEY ) ) {
+		return settings;
+	}
+	const existingGetEditWrapperProps = settings.getEditWrapperProps;
+	settings.getEditWrapperProps = ( attributes ) => {
+		let props = {};
+		if ( existingGetEditWrapperProps ) {
+			props = existingGetEditWrapperProps( attributes );
+		}
+		return addSaveProps( props, settings, attributes );
+	};
+
+	return settings;
+}
+
+const ColorPanel = ( { colorSettings, clientId } ) => {
+	const { getComputedStyle, Node } = window;
```

Comment:
I guess it’s not that different from what we had before but if you would see a way to extract a hook that operates on DOM and put it in its own file it might make it easier to replicate this behavior in React Native. Well, the best case scenario is it becomes no-op there 😃

Reply:
Yes, it's similar and I agree, it's seems like it could be its own hook.